staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -187,7 +184,8 @@ func (p *Patcher) buildMergePatch(original, modified, current []byte) ([]byte, e
 
 // buildStrategicMergeFromOpenAPI builds patch from OpenAPI if it is enabled.
 // This is used for core types which is published in openapi.
-func (p *Patcher) buildStrategicMergeFromOpenAPI(schema proto.Schema, original, modified, current []byte) ([]byte, error) {
+func (p *Patcher) buildStrategicMergeFromOpenAPI(original, modified, current []byte) ([]byte, error) {
+	schema := p.OpenapiSchema.LookupResource(p.Mapping.GroupVersionKind)
```

Comment:
What if the resource can't be found (i.e. schema is nil)? Will the code below not crash? Will it return an error?

Reply:
or when `p.OpenapiSchema` is nil

Reply:
> or when `p.OpenapiSchema` is nil

`p.OpenapiSchema` can not be nil, since it is checked on Line 121 before the `buildStrategicMergeFromOpenAPI` is called.

Reply:
> What if the resource can't be found (i.e. schema is nil)? Will the code below not crash? Will it return an error?

Added check for `nil` schema. In order to keep the functionality the same, this function should return a nil patch and a nil error.